acl/acl_test.go

```diff
@@ -781,6 +781,63 @@ func TestPolicyACL_Node(t *testing.T) {
 	}
 }
 
+func TestPolicyACL_KeyReadPrefix(t *testing.T) {
+	all := AllowAll()
+	policy := &Policy{
+		Keys: []*KeyPolicy{
+			&KeyPolicy{
+				Prefix: "",
+				Policy: PolicyDeny,
+			},
+			&KeyPolicy{
+				Prefix: "foo/",
+				Policy: PolicyWrite,
+			},
+			&KeyPolicy{
+				Prefix: "foo/priv/",
+				Policy: PolicyDeny,
+			},
+			&KeyPolicy{
+				Prefix: "bar/",
+				Policy: PolicyDeny,
+			},
+			&KeyPolicy{
+				Prefix: "bar/baz",
+				Policy: PolicyRead,
+			},
+			&KeyPolicy{
+				Prefix: "zip/",
+				Policy: PolicyRead,
+			},
+		},
+	}
+	acl, err := New(all, policy)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	type keycase struct {
+		inp        string
+		readPrefix bool
+	}
+	cases := []keycase{
+		{"other", false},
+		{"foo/test", true},
+		{"foo/priv/test", false},
+		{"bar/", true},
+		{"bar/deny", false},
+		{"foo/", true},
```

Comment:
This should be moved up and the next line is a duplicate of the above. Check "zip"?